pkg/policy/l4.go

```diff
@@ -168,65 +167,40 @@ func CreateL4IngressFilter(fromEndpoints api.EndpointSelectorSlice, rule api.Por
 		L7RulesPerEp:     make(L7DataMap),
 		Endpoints:        filterEndpoints,
 		DerivedFromRules: labels.LabelArrayList{ruleLabels},
+		Ingress:          ingress,
 	}
 
-	l4.Ingress = true
-
 	if protocol == api.ProtoTCP && rule.Rules != nil {
 		switch {
 		case len(rule.Rules.HTTP) > 0:
 			l4.L7Parser = ParserTypeHTTP
 		case len(rule.Rules.Kafka) > 0:
 			l4.L7Parser = ParserTypeKafka
 		}
-
-		l4.L7RulesPerEp.addRulesForEndpoints(*rule.Rules, fromEndpoints)
+		l4.L7RulesPerEp.addRulesForEndpoints(*rule.Rules, filterEndpoints)
```

Comment:
I believe that this is the right behaviour. We should ensure that we are setting up the endpointselectors in the L4Filter, rather than relying on layers further down to treat the empty list of peerEndpoints as a wildcard selector.

/cc @ianvernon